### ISSUE TITLE
Meta Data

### ISSUE THREAD

**Issue by RedNumberIX** [raft-tech/raft-tech.github.io#54]
There's a lot that goes into SEO, but one of the most important parts of it is our metadata. If we're going to write blogs, we've got to make sure Google knows how to show them to our potential visitors. 

Since we're not using an established CMS I'll need some way to insert the HTML code into the header of each page on the website (not only blogs but the main pages as well). Below are the tags I'm most interested in:
1. Meta Description
2. Canonical Tag
3. Alt Text Tags (for images)
4. Robots Meta Tag
5.  Open Graph Meta Tags

Better descriptions of the exact tags and code can be found here: https://clutch.co/seo-firms/resources/meta-tags-that-improve-seo

**Comment by BarakStout** [comment]
@RedNumberIX This is a meta-ticket and should be broken down or use a checklist and reference other tickets that can be worked independently. Adding a meta description and tags can be done rather quickly.

**Comment by RedNumberIX** [comment]
Sorry about that, I'm not too familiar with GitHub. I can break this down into smaller tasks but I have to ask a few questions regarding how exactly our CMS works. Right now I can't create any blogs myself. So it'll be you guys who upload this information for now. 
So how should I present this metadata for the blogs?
Should I provide all the metadata in the finalized copy along with the images?

And when I do take over these publishing duties how is this accomplished? 
Do we have some sort of tool that will autogenerate the HTML tags for the page automatically or will I have to write the code out myself by hand? I'm also assuming also that this process will be repeated for other pages on the website.

NOTE: Please, if you can make this a section on the Lunch-and-Learn!

**Comment by BarakStout** [comment]
@RedNumberIX see responses below. 

> Sorry about that, I'm not too familiar with GitHub. I can break this down into smaller tasks but I have to ask a few questions regarding how exactly our CMS works. Right now I can't create any blogs myself. So it'll be you guys who upload this information for now.

No problem. We try to divided one task per ticket. Take a look at https://github.com/raft-tech/raft-tech.github.io/issues/56 for an example. If it’s a large ticket, it can reference other tickets, we can touch on that at a later time. 

> So how should I present this metadata for the blogs?

> Should I provide all the metadata in the finalized copy along with the images?

The meta data for the blogs will be added to the Front Matter section of each post. Take a look at the first few lines of this file https://raw.githubusercontent.com/raft-tech/raft-tech.github.io/master/_posts/2020-03-10-ui-design-trends.md any meta data would be added in the first few lines and rendered to html tags as the site gets generated. 

> And when I do take over these publishing duties how is this accomplished?

There is this ticket https://github.com/raft-tech/raft-tech.github.io/issues/59 to add auto post, so ideally no one will have this duties. As a manual trigger, things are published what the are merged into the repo. Basically a click on a button. 

> Do we have some sort of tool that will autogenerate the HTML tags for the page automatically or will I have to write the code out myself by hand? I'm also assuming also that this process will be repeated for other pages on the website.

You shouldn’t need to write any HTML unless you are editing the layout files.

**Comment by BarakStout** [comment]
1. Use `excerpt` in the front matter of any post or page and the site will generate the description tag (truncated at 160 chars) 
2. Canonical tags are already generated
3. Already done 
4. After #92 gets merged, use `robots` in the front matter of any post or page and the site will generate the robots tag
5. Open Graph Meta Tags are already generated 

<img width="458" alt="Screen Shot 2020-04-03 at 8 33 09 AM" src="https://user-images.githubusercontent.com/34528865/78361170-f59c9000-7585-11ea-8e97-5c86ca5a55f9.png">
<img width="341" alt="Screen Shot 2020-04-03 at 8 33 02 AM" src="https://user-images.githubusercontent.com/34528865/78361180-f9301700-7585-11ea-875e-1011864c7c41.png">
<img width="512" alt="Screen Shot 2020-04-03 at 8 32 44 AM" src="https://user-images.githubusercontent.com/34528865/78361114-d69dfe00-7585-11ea-9e54-632a85d0e9fb.png">
